src/main/webapp/app/core/course/overview/exercise-details/course-exercise-details.component.ts

```diff
@@ -272,18 +272,19 @@ export class CourseExerciseDetailsComponent implements OnInit, OnDestroy {
                 .flatMap(
                     (participation) =>
                         participation.submissions?.flatMap((submission) => {
-                            submission.results?.forEach((result) => {
-                                result.submission = submission;
-                            });
-                            submission.participation = participation;
-                            return submission.results ?? [];
+                            return (
+                                submission.results?.map((result) => {
+                                    result.submission = submission;
+                                    result.submission.participation = participation;
+                                    return result;
+                                }) ?? []
+                            );
```

Comment:
I do not get the difference here; why did we change it? What is the advantage of the new approach?